helpers/search_processing_kendra_quickstart.sh

```diff
@@ -0,0 +1,419 @@
+kendra_intelligent_ranking.aws.secret_key:$AWS_SECRET_ACCESS_KEY
+EOF
+  if [ -n "${AWS_SESSION_TOKEN:-}" ]; then
+    # If a session token was found in the environment / profile, push it to the OpenSearch keystore too.
+    echo "WARNING: Adding session token to OpenSearch keystore. These temporary credentials will"
+    echo "expire, and should only be used for short-lived testing."
+    cat >>kendra_ranking.credentials <<EOF
+kendra_intelligent_ranking.aws.session_token:$AWS_SESSION_TOKEN
+EOF
+  fi
+  cat >install_credentials.sh <<"EOF"
+#!/bin/bash
+
+for l in $(cat $1); do
+  KEY=$(echo $l | cut -f1 -d:)
+  VALUE=$(echo $l | cut -f2 -d:)
+  echo $VALUE | /usr/share/opensearch/bin/opensearch-keystore add $KEY --stdin
+done
+EOF
+  chmod 755 install_credentials.sh
+  cat >>Dockerfile << EOF
+
+# Push credentials to keystore
+COPY --chown=opensearch:opensearch install_credentials.sh /tmp
+COPY --chown=opensearch:opensearch kendra_ranking.credentials /tmp
+RUN /usr/share/opensearch/bin/opensearch-keystore create
+RUN --mount=type=secret,id=credentials,target=/tmp/kendra_ranking.credentials,required=true,mode=0444 \
+  /tmp/install_credentials.sh /tmp/kendra_ranking.credentials
+EOF
+fi
+
+# 
+# Build and tag the Docker image with the plugin (and maybe credentials in the keystore)
+#
+if [ -f kendra_ranking.credentials ]; then
+  DOCKER_BUILDKIT=1 docker build --tag ${DOCKER_IMAGE_TAG} --secret id=credentials,src=kendra_ranking.credentials .
+  rm kendra_ranking.credentials
+else
+  docker build --tag ${DOCKER_IMAGE_TAG} .
+fi
+
+#
+# Make sure we have opensearch-dashboards:
+#
+docker pull ${OPENSEARCH_DASHBOARDS_IMAGE_TAG}
+
+# 
+# Create a docker-compose.yml file that will launch an OpenSearch node with the image we
+# just built and an OpenSearch Dashboards node that points to the OpenSearch node.
+#
+cat >docker-compose.yml <<EOF
+version: '3'
+networks:
+  opensearch-net:
+services:
+  opensearch-node:
+    image: ${DOCKER_IMAGE_TAG}
+    container_name: opensearch-node
+    environment: 
+      - cluster.name=opensearch-cluster
+      - node.name=opensearch-node
+      - discovery.type=single-node
+      - kendra_intelligent_ranking.service.endpoint=${KENDRA_RANKING_ENDPOINT}
+      - kendra_intelligent_ranking.service.region=${AWS_REGION}
+      - kendra_intelligent_ranking.service.execution_plan_id=${EXECUTION_PLAN_ID}
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+      nofile:
+        soft: 65536
+        hard: 65536
+    ports:
+      - 9200:9200
+      - 9600:9600
+    networks:
+      - opensearch-net
+  opensearch-dashboard:
+    image: ${OPENSEARCH_DASHBOARDS_IMAGE_TAG}
+    container_name: opensearch-dashboards
+    ports:
+      - 5601:5601
+    environment:
+      OPENSEARCH_HOSTS: '["https://opensearch-node:9200"]'
+    networks:
+      - opensearch-net
+EOF
+
+# 
+# Bring up the images
+#
+docker-compose up -d
+
+#
+# Output helpful "getting started" text.
+#
+cat >README <<EOF
```

Comment:
Also include instruction for Cleanup
- remove docker containers
- delete execution plan

Actually - much better to add an option to the script to completely cleanup all resources automatically.. Add confirmation (Yes/No) to avoid accidental deletion.

Reply:
The script now generates its own cleanup script that shuts down Docker containers and deletes any created rescore execution plan.

The output looks like:

```
This script will shut down and clean up running Docker containers for OpenSearch and OpenSearch Dashboards.
And it will delete the Kendra Intelligent Ranking rescore execution plan with ID d08cb653-4a51-4f38-b765-75cd767b21f8
Are you sure? (yes/no)
yes
Stopping opensearch-node       ... done
Stopping opensearch-dashboards ... done
Removing opensearch-node       ... done
Removing opensearch-dashboards ... done
Removing network opensearch-kendra-ranking-dockertino_opensearch-net
Deleting rescore execution plan with ID d08cb653-4a51-4f38-b765-75cd767b21f8...
```

It will only delete stuff with an explicit `yes` answer:

```
This script will shut down and clean up running Docker containers for OpenSearch and OpenSearch Dashboards.
And it will delete the Kendra Intelligent Ranking rescore execution plan with ID d08cb653-4a51-4f38-b765-75cd767b21f8
Are you sure? (yes/no)
y
Are you sure? (yes/no)
maybe
Are you sure? (yes/no)
sometimes
Are you sure? (yes/no)
no
```